base/sparse/sparsematrix.jl

```diff
@@ -206,7 +206,7 @@ function sparse_compute_reshaped_colptr_and_rowval{Ti}(colptrS::Vector{Ti}, rowv
     end
 end
 
-function reinterpret{T,Tv,Ti,N}(::Type{T}, a::SparseMatrixCSC{Tv,Ti}, dims::NTuple{N,Int})
+function reinterpret(::Type{T}, a::SparseMatrixCSC{Tv,Ti}, dims::NTuple{N,Int}) where {T,Tv,Ti,N}
```

Comment:
When long-form method definitions involve several type parameters, the compact style in this definition (`where {T,Tv,Ti,N}`) seems clearly preferable to `where T where Tv where Ti where N`. Consistency of style also being advantageous, after what number of type parameters should the compact style be preferred in long-form method definitions? Drawing the line after anything more than one type parameter seems a bit arbitrary. Thoughts?